job/auth.go

```diff
@@ -0,0 +1,81 @@
+package job
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	jwtverifier "github.com/okta/okta-jwt-verifier-golang"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+	"golang.org/x/oauth2"
+)
+
+var (
+	username = ""
+	password = ""
+
+	Oauth2Config *oauth2.Config           = nil
+	Verifier     *jwtverifier.JwtVerifier = nil
+)
+
+func SetupAuth(issuer string, audience string, clientId string, clientSecret string, userName string, pwd string,
+	tokenUrl string) {
+}
+
+func InitAuth() {
```

Comment:
you could do this in magic module method init() and go calls that method while loading the module. There's no need to call explicitly & can make this private scoped

Reply:
I tried that first and it initialized before it had any configuration.